front/src/applications/editor/tools/switchEdition/components.tsx

```diff
@@ -207,14 +206,23 @@ export const SwitchEditionLeftPanel: FC = () => {
     [switchEntity.properties.switch_type, switchTypes]
   );
   const flatSwitchEntity = useMemo(
-    () => switchToFlatSwitch(switchType, switchEntity),
+    () => switchType && switchToFlatSwitch(switchType, switchEntity),
     [switchEntity, switchType]
   );
   const switchTypeJSONSchema = useMemo(
-    () => baseSchema && getSwitchTypeJSONSchema(baseSchema, switchType),
+    () => switchType && baseSchema && getSwitchTypeJSONSchema(baseSchema, switchType),
     [baseSchema, switchType]
   );
 
+  /**
+   * FIX crash if the user change the current infra while editing a switch
+   * https://github.com/osrd-project/osrd/issues/4801
+   * No need to display the component if there no switches to display.
+   */
+  if (!switchTypes || switchTypes.length === 0) {
```

Comment:
Fix switchType by removing the cast in SwitchType and rather check that switchType is not undefined or null
```suggestion
  if (!switchType || !flatSwitchEntity) {
```

Reply:
yes, more clean